### PR TITLE
Documentation for the read replicas

### PR DESCRIPTION
**Pull request by fowczarek**
Documentation for the read replicas

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4526 Implement database read replicas](https://app.clickup.com/t/2549495/SALEOR-4526)

File: docs/developer/running-saleor/read-replicas.mdx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+In `DATABASES` we should add new entry:
+`DATABASE_CONNECTION_REPLICA_NAME: dj_database_url.config(default="postgres://saleor:saleor@localhost:5433/saleor",conn_max_age=600)`,
+
+
+## How to use replica in code
+
+By default Saleor code always use the primary database.
+
+To specify which connection should be used by ORM we use [`using` on queryset.](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/#manually-selecting-a-database)
```
Comment (maarcingebala) [inline docs/developer/running-saleor/read-replicas.mdx:38]:
```suggestion
To specify which connection to use in ORM, use the [`using` queryset method.](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/#manually-selecting-a-database)
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+> **Example**
+>
+> To fetch all products from the primary database we could use:
```
Comment (maarcingebala) [inline docs/developer/running-saleor/read-replicas.mdx:42]:
```suggestion
> To fetch all products from the primary database use:
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+>
+> - Products.objects.all()
+> - Products.objects.using(DATABASE_CONNECTION_DEFAULT_NAME).all()
+>
+> To fetch all products from the read replica we could use:
```
Comment (maarcingebala) [inline docs/developer/running-saleor/read-replicas.mdx:47]:
```suggestion
> To fetch all products from the read replica use:
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+>
+> - Products.objects.using(DATABASE_CONNECTION_REPLICA_NAME).all()
+
+:::note
+Do not forget about importing the proper connection name from settings.
```
Comment (maarcingebala) [inline docs/developer/running-saleor/read-replicas.mdx:52]: We could also add this import in the snippets above so it is clear where do these values come from.

File: docs/developer/running-saleor/read-replicas.mdx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+`database_connection_name` as a part of the base `Dataloader` class. All dataloaders ORM queries should use this variable as connection name
+> **Example**
+>
+>
+> Product.objects.using(self.database_connection_name).in_bulk(keys)
+
+
+## How to test it in a local machine
+
+To simulate read replica on the local machine we could use the second connection to the primary database with the read-only user.
```
Comment (maarcingebala) [inline docs/developer/running-saleor/read-replicas.mdx:68]:
```suggestion
To simulate a read replica on the local machine, use the second connection to the primary database with the read-only user.
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+
+### How to use read replicas in dataloaders
+
+Dataloaders could be used to make responses for mutations and queries. Mutations should always return data from the primary database to show real-time data as a result of mutation. To handle this situation we introduce
+`database_connection_name` as a part of the base `Dataloader` class. All dataloaders ORM queries should use this variable as connection name
+> **Example**
+>
+>
+> Product.objects.using(self.database_connection_name).in_bulk(keys)
```
Comment (krzysztofwolski) [inline docs/developer/running-saleor/read-replicas.mdx:63]:
```suggestion
`database_connection_name` as a part of the base `Dataloader` class. All dataloaders ORM queries should use this variable as connection name, for example:
```python
    Product.objects.using(self.database_connection_name).in_bulk(keys)
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+
+## How to test it in a local machine
+
+To simulate read replica on the local machine we could use the second connection to the primary database with the read-only user.
+
+To setup it we should add read-only user to database
```
Comment (maarcingebala) [inline docs/developer/running-saleor/read-replicas.mdx:70]: 1. Add a read-only user in your database:
```
SQL example to create the user
```
If you're using the [saleor-platform](link), you will have a read-only user created automatically.

2. Add the following configuration in `settings.py`:

```
full example from settings.py
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    DATABASE_CONNECTION_REPLICA_NAME: dj_database_url.config(
+        default=REPLICA_DB_URL, conn_max_age=600,
+    ),
+}
+```
+
+
+## How to use replica in code
+
+By default Saleor always uses the primary database. To specify which connection to use in ORM, use [the `using` queryset method.](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/#manually-selecting-a-database)
```
Comment (krzysztofwolski) [inline docs/developer/running-saleor/read-replicas.mdx:44]:
```suggestion
By default Saleor always uses the primary database. To specify which connection to use in ORM, use [the `using` queryset method](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/#manually-selecting-a-database).
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]
Only lil thing and LGTM

**Review by maarcingebala — APPROVED** [review]
😍

**Review by tomaszszymanski129 — APPROVED** [review]

File: docs/developer/running-saleor/read-replicas.mdx [hunk 9]
```diff
@@ -0,0 +1,112 @@
+---
+title: Database read replicas
+---
+
+Database read replica is a read-only copy of the primary database. Replica should reflect changes in the primary database as fast as possible. This feature allows Saleor to offload the primary database by moving read-only
```
Comment (karolkielecki) [inline docs/developer/running-saleor/read-replicas.mdx:5]:
```suggestion
Database read replica is a read-only copy of the primary database. Replica should reflect changes in the primary database as fast as possible. This feature allows Saleor to offload the primary database by moving the read-only traffic to a separate database.
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+traffic to a separate database.
```
Comment (karolkielecki) [inline docs/developer/running-saleor/read-replicas.mdx:6]:
```suggestion

```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+ Product.objects.using(self.database_connection_name).in_bulk(keys)
+```
+
+:::note
+Mutations should always use data from the primary database to ensure that they operate on the newest data.
+:::
+
+## How to test it in a local machine
+
+To simulate a read replica on the local machine, use the second connection to the primary database with the read-only user.
```
Comment (karolkielecki) [inline docs/developer/running-saleor/read-replicas.mdx:79]:
```suggestion
To simulate a read replica on a local machine, use a second connection to the primary database with the read-only user.
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+```mermaid
+graph LR
+  C((Saleor Core)) ---|Read/Write| P[(Primary database)]
+  C ---|Read| R[(Read replica)]
+  P --->|Asynchronous replication| R
+```
+
+## How to configure
+Django allows using [multiple databases.](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/)
+To start using read replica in Saleor set `DATABASE_CONNECTION_REPLICA_NAME` in `settings.py` and specify this connection in `DATABASES`.
```
Comment (karolkielecki) [inline docs/developer/running-saleor/read-replicas.mdx:19]:
```suggestion
To start using read replica in Saleor, set `DATABASE_CONNECTION_REPLICA_NAME` in `settings.py` and specify this connection in `DATABASES`.
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+    DATABASE_CONNECTION_REPLICA_NAME: dj_database_url.config(
+        default=REPLICA_DB_URL, conn_max_age=600,
+    ),
+}
+```
+
+
+## How to use replica in code
+
+By default Saleor always uses the primary database. To specify which connection to use in ORM, use [the `using` queryset method](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/#manually-selecting-a-database).
```
Comment (karolkielecki) [inline docs/developer/running-saleor/read-replicas.mdx:44]:
```suggestion
Saleor always uses the primary database by default. To specify which connection to use in ORM, use [the `using` queryset method](https://docs.djangoproject.com/en/4.0/topics/db/multi-db/#manually-selecting-a-database).
```

File: docs/developer/running-saleor/read-replicas.mdx [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+#### 1. Add a read-only user in your database:
+```sql
+CREATE USER saleor_read_only WITH PASSWORD 'saleor';
+GRANT CONNECT ON DATABASE saleor TO saleor_read_only;
+GRANT USAGE ON SCHEMA public TO saleor_read_only;
+GRANT SELECT ON ALL TABLES IN SCHEMA public TO saleor_read_only;
+ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT ON TABLES TO saleor_read_only;
+```
+:::note
+Saleor-platform creates read-only user by default.
```
Comment (karolkielecki) [inline docs/developer/running-saleor/read-replicas.mdx:90]:
```suggestion
Saleor-platform creates a read-only user by default.
```

**Review by karolkielecki — CHANGES_REQUESTED** [review]

**Review by karolkielecki — APPROVED** [review]

**Pull request merged**